sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -0,0 +1,183 @@
+
+@Injectable()
+export class AssessmentSequentialProcessingService {
+  /**
+   * Checks if changes in an assessment can potentially causes changes in another application
+   * which would demand a reassessment of the same.
+   * @param assessmentId assessment currently changing (e.g. updated or cancelled).
+   * @param auditUserId user that should be considered the one that is causing the changes.
+   * @param entityManager used to execute the commands in the same transaction.
+   * @returns impacted application if any, otherwise null.
+   */
+  async assessImpactedApplicationReassessmentNeeded(
+    assessmentId: number,
+    auditUserId: number,
+    entityManager: EntityManager,
+  ): Promise<Application | null> {
+    const applicationRepo = entityManager.getRepository(Application);
+    // Application which current assessment is the assessmentId to be checked.
+    // If the assessment id is not associated with the current application's assessment
+    // no need to check for impacts on future applications.
+    const application = await applicationRepo.findOne({
+      select: {
+        id: true,
+        applicationNumber: true,
+        programYear: {
+          id: true,
+        },
+        student: {
+          id: true,
+        },
+      },
+      relations: {
+        student: true,
+        programYear: true,
+      },
+      where: {
+        applicationStatus: Not(ApplicationStatus.Overwritten),
+        currentAssessment: {
+          id: assessmentId,
+        },
+      },
+    });
+    if (!application) {
+      // Assessment is not current one or the application is overwritten.
+      return null;
+    }
+    const sequencedApplications = await this.getSequencedApplications(
+      application.applicationNumber,
+      application.student.id,
+      application.programYear.id,
+      entityManager,
+    );
+    if (!sequencedApplications.future.length) {
+      // There are no future applications impacted.
+      return null;
+    }
+    // Create the reassessment for the next impacted application.
+    const [futureSequencedApplication] = sequencedApplications.future;
+    const impactedApplication = new Application();
+    impactedApplication.id = futureSequencedApplication.applicationId;
+    const now = new Date();
+    // Create the new assessment to be processed.
+    const auditUser = { id: auditUserId } as User;
+    impactedApplication.currentAssessment = {
+      application: {
+        id: impactedApplication.id,
+      } as Application,
+      offering: {
+        id: futureSequencedApplication.currentAssessmentOfferingId,
+      },
+      triggerType: AssessmentTriggerType.RelatedApplicationChanged,
+      relatedApplicationAssessment: { id: assessmentId } as StudentAssessment,
+      creator: auditUser,
+      createdAt: now,
+      submittedBy: auditUser,
+      submittedDate: now,
+    } as StudentAssessment;
+    return applicationRepo.save(impactedApplication);
```

Comment:
impactedApplication.modifier and updateAt?

Reply:
Nice catch, added 😉